disnake/interactions/base.py

```diff
@@ -209,7 +209,9 @@ def channel(self) -> Union[TextChannel, Thread]:
         channel = guild and guild._resolve_channel(self.channel_id)
         if channel is None:
             if self.channel_id is not None:
-                type = ChannelType.text if self.guild_id is not None else ChannelType.private
+                type = (
+                    None if self.guild_id is not None else ChannelType.private
+                )  # could be a text, voice, or thread channel in a guild
```

Comment:
it could be either voice or text, and I don't think returning None here is the answer.

Reply:
If we can't resolve the channel and the channel ID is provided by Discord, there's no way to tell exactly what channel type the interaction originated from. I don't believe there *is* an answer other than returning None - the PartialMessageable typing certainly allows for it, and any other value would only be an assumption.
Judging by the other comments in this code, this line handles the case when the interaction originates from a DM anyway (or potentially when running a bot with very limited intents, such that the guild cache is not populated?).

Reply:
Looking at how this is used, I suppose it looks fine